spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/FindBadEndOfStreamCheckTest.java

```diff
@@ -0,0 +1,160 @@
+package edu.umd.cs.findbugs.detect;
+
+import static edu.umd.cs.findbugs.test.CountMatcher.containsExactly;
+import static org.hamcrest.Matchers.hasItem;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+
+import edu.umd.cs.findbugs.AbstractIntegrationTest;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcher;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcherBuilder;
+
+public class FindBadEndOfStreamCheckTest extends AbstractIntegrationTest {
+    @Test
+    public void testBadEndOfFileChecks() {
+        performAnalysis("endOfStreamCheck/BadEndOfStreamCheck.class");
+
+        // There should be exactly 16 issue of this type
+        final BugInstanceMatcher bugTypeMatcher = new BugInstanceMatcherBuilder()
+                .bugType("EOS_BAD_END_OF_STREAM_CHECK").build();
+        assertThat(getBugCollection(), containsExactly(16, bugTypeMatcher));
+
+        BugInstanceMatcher bugInstanceMatcher = new BugInstanceMatcherBuilder()
```

Comment:
Perhaps extract the bug instance assembly and check to a helper method, since several parts of it are repeated?